File: typescript/examples/07-good-law-cited-by.ts
/**
 * Example 07 — Good-law detail + cited-by pagination
 * Language: TypeScript (Node.js)
 *
 * Usage: npx tsx examples/07-good-law-cited-by.ts [caseId]
 * Default caseId is United States v. Windsor (570 U.S. 744) as used in the docs.
 */
import { LawDiverClient } from "../src/client.js";

const client = new LawDiverClient();
const caseId = process.argv[2] ?? "2812209";

const goodLaw = await client.goodLaw(caseId);
console.log("=== Good-law ===");
console.log(JSON.stringify(goodLaw, null, 2).slice(0, 1500));
console.log("");

const citedBy = (await client.citedBy(caseId, { limit: 10, offset: 0 })) as {
  results?: unknown[];
  total?: number;
  requestId?: string;
};
console.log("=== Cited by (first page) ===");
console.log(`total≈ ${citedBy.total ?? "?"} · requestId=${citedBy.requestId}`);
const rows = citedBy.results ?? (citedBy as unknown as { cases?: unknown[] }).cases ?? [];
for (const row of rows.slice(0, 10)) {
  const r = row as { caseName?: string; citation?: string; year?: number };
  console.log(`• ${r.caseName ?? JSON.stringify(row).slice(0, 100)} — ${r.citation ?? ""} (${r.year ?? ""})`);
}

const meta = await client.caseMetadata(caseId);
console.log("\n=== Metadata (truncated) ===");
console.log(JSON.stringify(meta, null, 2).slice(0, 800));

File: python/examples/07_good_law_cited_by.py
"""Example 07 — Good-law + cited-by (Python)."""

from __future__ import annotations

import json
import sys
from pathlib import Path

sys.path.insert(0, str(Path(__file__).resolve().parents[1]))

from lawdiver import LawDiverClient

case_id = sys.argv[1] if len(sys.argv) > 1 else "2812209"

with LawDiverClient() as client:
    good = client.good_law(case_id)
    print("=== Good-law ===")
    print(json.dumps(good, indent=2)[:1500])
    print()

    cited = client.cited_by(case_id, limit=10, offset=0)
    print("=== Cited by (first page) ===")
    print(f"total≈ {cited.get('total', '?')} · requestId={cited.get('requestId')}")
    rows = cited.get("results") or cited.get("cases") or []
    for row in rows[:10]:
        print(f"• {row.get('caseName')} — {row.get('citation')} ({row.get('year')})")

    meta = client.case_metadata(case_id)
    print("\n=== Metadata (truncated) ===")
    print(json.dumps(meta, indent=2)[:800])
